### ISSUE TITLE
Order confirmation email send before the payment done

### ISSUE THREAD

**Issue by spras** [PayboxByVerifone/Magento2#10]
The order confirmation email is send before the payment is done.

This is because in Paybox\Epayment\Model\Payment\AbstractPayment 
```
public function getOrderPlaceRedirectUrl()
    {
        return $this->getUrl('pbxep/payment/redirect', array('_secure' => true));
    }
```
returns null

**Comment by max21** [comment]
Do you have resolved this issue ?

**Comment by pixmin** [comment]
On a default Magento 2.2.5 (but I believe this is standard on all Magento 2 versions), when I choose "Check / Money order" as a payment method, I do receive an order confirmation.

The next step is to validate the payment (manually in this case, or via IPN when using Paybox). This will send an Invoice for the order. Those are two different email confirmations.

@spras Why do you believe the order confirmation should not be sent before the payment is done?

**Comment by coordtechjetpulp** [comment]
Because, it seems weird to receive an order confirmation email when the payment failed. 
And i don't remember receiving a payment confirmation email, in case of success ?

**Comment by spras** [comment]
Because, it seems weird to receive an order confirmation email when the payment failed.
And i don't remember receiving a payment confirmation email, in case of success ?

**Comment by pixmin** [comment]
You should definitely receive an email confirmation (Invoice email) when:

- You manually confirm the order
- The IPN is received from the payment platform

Maybe you didn't receive the email because the IPN didn't get through?

**Comment by spras** [comment]
Don't remember, it was more than one year ago... i think it  weird for users to receive two confirmation emails in case of success et one confirmation email in case of payment failure.

**Comment by pixmin** [comment]
Agreed, but that's how Magento works, nothing to do with Paybox as far as I know since ordering by check does the very same thing...
